cmd/immuadmin/command/backup_test.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2019-2020 vChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package immuadmin
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	immuclient "github.com/codenotary/immudb/pkg/client"
+	immudb "github.com/codenotary/immudb/pkg/server"
+)
+
+func generateRandomTCPPort() int {
```

Comment:
@padurean  Actually I planned to refactor generateRandomTCPPort a bit to make sure it's randomness works better.
gRPC buffcon package is a good alternative, at that time of writing the test my thought was that starting a server instance would make tests more appear as a real life interaction and that 64000 ports would be enough for the tests.

Howerver if needed I can refactor the tests to use gRPC buffcon

Reply:
From my point of view we should never start real servers in unit tests.
But i would like a second opinion on this.
@volknanebo @mmeloni What's your take on this?

Reply:
Other disadvantages are:
- servers starting up take more time => tests will get slower and slower with this approach
- possible side effects - e.g. this [comment] (https://github.com/codenotary/immudb/blob/master/pkg/gw/safeset_handler_test.go#L47) about metrics server
- the need to use `time.Sleep` in tests which also makes them take longer and with time it can add significant delays